src/firestore/delete.js

```diff
@@ -26,21 +26,28 @@ var MIN_ID = "__id-9223372036854775808__";
  */
 function FirestoreDelete(project, path, options) {
   this.project = project;
-  this.path = path;
+  this.path = path || "";
   this.recursive = Boolean(options.recursive);
   this.shallow = Boolean(options.shallow);
   this.allCollections = Boolean(options.allCollections);
 
   // Remove any leading or trailing slashes from the path
-  if (this.path) {
-    this.path = this.path.replace(/(^\/+|\/+$)/g, "");
-  }
-
-  this.isDocumentPath = this._isDocumentPath(this.path);
-  this.isCollectionPath = this._isCollectionPath(this.path);
+  this.path = this.path.replace(/(^\/+|\/+$)/g, "");
 
   this.allDescendants = this.recursive;
-  this.parent = "projects/" + project + "/databases/(default)/documents";
+  this.root = "projects/" + project + "/databases/(default)/documents";
+
+  var segments = this.path.split("/");
+  this.isDocumentPath = segments.length % 2 === 0;
+  this.isCollectionPath = !this.isDocumentPath;
+
+  this.parent = this.root;
```

Comment:
I think this block here could use an explanatory comment since I am having some trouble running it in my head. In particular the `pop()` followed by the zero-length checking is hard to interpret.

Reply:
Some examples to make sure I have it right:

Input: `path=/users`
Result: `parent=/`

Input: `path=/users/sam`
Result: `parent=/users/sam`

Input: `path=/users/sam/items`
Result: `parent=/users/sam`

Reply:
So in a sentence: the "parent" for a collection is the document that owns it, or the root.  And the "parent" for a document is itself.

Reply:
Gotcha. "parent" is a pretty common concept in the backend, we should consider making document names a first-class citizen (maybe hijack some code from the Node.js SDK? maybe even move this logic into the Node.js SDK and pull it into the CLI tool as a dependency? @hiranya911 thoughts?)